src/lib/y2packager/dialogs/inst_product_license.rb

```diff
@@ -0,0 +1,147 @@
+      #
+      # This happens when the user changes the license language
+      def language_handler
+        self.language = Yast::UI.QueryWidget(Id(:language), :Value)
+        Yast::UI.ReplaceWidget(Id(:license_replace_point), license_content)
+      end
+
+      # Handler for the :license_confirmation action
+      #
+      # This action happens when the user clicks the confirmation checkbox.
+      def license_confirmation_handler
+        @confirmed = Yast::UI.QueryWidget(Id(:license_confirmation), :Value)
+      end
+
+      # Handler for the :next action
+      #
+      # This action happens when the user clicks the 'Next' button
+      def next_handler
+        if confirmed
+          update_product_confirmation
+          finish_dialog(:next)
+        else
+          Yast::Report.Message(_("You must accept the license to install this product"))
+        end
+      end
+
+      # Handler for the :next action
+      #
+      # This action happens when the user clicks the 'Back' button
+      def back_handler
+        update_product_confirmation
+        finish_dialog(:back)
+      end
+
+    private
+
+      # @return [String] Language code (en_US, es_ES, etc.).
+      attr_accessor :language
+      # @return [Boolean] Determines whether the user confirmed the license
+      attr_accessor :confirmed
+
+      # Dialog content
+      #
+      # @see ::UI::Dialog
+      def dialog_content
+        VBox(
+          Left(language_selection),
+          VSpacing(0.5),
+          ReplacePoint(
+            Id(:license_replace_point),
+            license_content
+          ),
+          confirmation_checkbox
+        )
+      end
+
+      # Dialog title
+      #
+      # @see ::UI::Dialog
+      def dialog_title
+        format(_("%s License Agreement"), product.label)
+      end
+
+      # Return the UI for the language selector
+      def language_selection
+        ComboBox(
+          Id(:language),
+          Opt(:notify, :hstretch),
+          _("&Language"),
+          Yast::Language.GetLanguageItems(:primary)
+        )
+      end
+
+      # Return the UI for the license content
+      def license_content
+        MinWidth(
+          80,
+          RichText(Id(:license_content), product.license(language))
+        )
+      end
+
+      # Return the UI for the confirmation checkbox
+      def confirmation_checkbox
+        VBox(
+          VSpacing(0.5),
+          Left(
+            CheckBox(
+              Id(:license_confirmation),
+              Opt(:notify),
+              # license agreement check box label
+              _("I &Agree to the License Terms."),
+              confirmed
+            )
+          )
+        )
+      end
+
+      # Update the product's license confirmation status
+      #
+      # It will not update the status if it has not changed.
+      def update_product_confirmation
+        return if product.license_confirmed? == confirmed
+        if confirmed
+          product.confirm_license
```

Comment:
Maybe the API could be `confirm_license(confirm = true)` so you could simply call `product.confirm_license(confirmed)` instead of the condition. Or have a `confirm_license`/`deny_license`/`set_license(bool)` combo...